### DISCUSSION TITLE
DataHub -> Data Objects Importer

### DISCUSSION THREAD

**Discussion by ismetsener** [pimcore/pimcore#9787] [Q&A / Support] [answered]
Hello,

I configured in "DataHub -> Data Objects Importer"  to import  objects
objectname is "DomainName"
one of the object field is a  "expiration_date"    (date format)

in excel.csv  I  defined date as "2020-01-01 00:00:00"
but when I try to import, I get a error:

```
In DomainName.php line 231
row 231  is like:
public function setExpiration_date(?\Carbon\Carbon $expiration_date)
```

```
Pimcore\Model\DataObject\DomainName::setExpiration_date(): Argument #1 ($expiration_date) must be of type ?Carbon\Carbon, DateTime given, called in /opt/pimcore/vendor/pimcore/data-importer/src/Mapping/
  DataTarget/Direct.php on line 50

```

How to define the dateformat in excel so that it imports without problem?

**Comment by weisswurstkanone** [comment]
Absolutely not sure but may a Carbon should be created here ?
https://github.com/pimcore/data-importer/blob/40676d773d839681c3b0148a669ca148a0b29a03/src/Mapping/Operator/Factory/Date.php#L44

**Comment by ismetsener** [comment]
not clear to me what to do?

**Reply by weisswurstkanone** [reply]
It was just a gut feeling. 
The question would be: 
Would it be possible on your side to investigate this further or confirm it. In the latter case it would be easy to fix. PRs are welcome.
Thanks for your feedback :-)

**Reply by ismetsener** [reply]
I am not  a developer, but I wil do my best to find a solution (or workaround)

**Comment by ismetsener** [comment, marked as answer]
Hi
thanks for your help.
I changed line 44  in   `/opt/pimcore/vendor/pimcore/data-importer/src/Mapping/Operator/Factory/Date.php`

`  $data = \DateTime::createFromFormat($this->format, $data);`
   in 
` $data = \Carbon\Carbon::createFromFormat($this->format, $data);`

Problem is Solved
tanks for helping me to the right direction.

**Reply by dvesh3** [reply]
@ismetsener could you please provide a PR on https://github.com/pimcore/data-importer? It should be quite easy, since this is a single file fix, so you can just edit the file directly on Github.

**Reply by fashxp** [reply]
already there? https://github.com/pimcore/data-importer/pull/62

**Reply by ismetsener** [reply]
> @ismetsener could you please provide a PR on https://github.com/pimcore/data-importer? It should be quite easy, since this is a single file fix, so you can just edit the file directly on Github.

I have no experience with Github,
what steps I have to take to change this?
what is 'PR" ?

**Reply by dvesh3** [reply]
@ismetsener A PR is a pull request for changes you want to merge on a Github code repository. Never mind, like fashxp mentioned this is already done pimcore/data-importer#62
